src/app/test/test-table/test-table.component.ts

```diff
@@ -0,0 +1,160 @@
+
+@Component({
+  selector: 'app-test-table',
+  standalone: true,
+  imports: [FormsModule],
+  templateUrl: './test-table.component.html',
+  styleUrl: './test-table.component.css',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TestTableComponent implements OnChanges, OnInit, OnDestroy {
+  @Input() reports: TestListItem[] = new Array<TestListItem>();
+  @Input() currentFilter: string = '';
+  @Input() showStorageIds?: boolean;
+  @Output() toggleCheckEvent: EventEmitter<TestListItem> = new EventEmitter<TestListItem>();
+  @Output() runEvent: EventEmitter<TestListItem> = new EventEmitter<TestListItem>();
+  protected amountOfSelectedReports: number = 0;
+  private subscriptions: Subscription = new Subscription();
+
+  constructor(
+    private httpService: HttpService,
+    private errorHandler: ErrorHandling,
+    private tabService: TabService,
+    private helperService: HelperService,
+    private testReportsService: TestReportsService,
+    private toastService: ToastService,
+  ) {}
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes['currentFilter'] || changes['reports']) {
+      this.amountOfSelectedReports = 0;
+      for (const report of this.reports) {
+        if (report.checked) {
+          this.amountOfSelectedReports++;
+        }
+        if (report.variables) {
+          report.extractedVariables = this.extractVariables(report.variables);
+        }
+      }
+      this.testReportsService.setAmountSelected(this.amountOfSelectedReports);
+      this.getFullPaths();
+    }
+  }
+
+  ngOnInit(): void {
+    this.subscribeToSubscriptions();
+  }
+
+  ngOnDestroy(): void {
+    this.unsubscribeToSubscriptions();
+  }
+
+  subscribeToSubscriptions(): void {
+    const amountSelectedSubscription: Subscription = this.testReportsService.amountSelected$.subscribe({
+      next: (value: number): void => {
+        this.amountOfSelectedReports = value;
+      },
+      error: () => catchError(this.errorHandler.handleError()),
+    });
+    this.subscriptions.add(amountSelectedSubscription);
+  }
+
+  unsubscribeToSubscriptions(): void {
+    this.subscriptions.unsubscribe();
+  }
+
+  openReport(storageId: number): void {
+    this.httpService.getReport(storageId, this.testReportsService.storageName).subscribe({
+      next: (report: Report): void => {
+        const reportData: ReportData = {
+          report: report,
+          currentView: {
+            storageName: this.testReportsService.storageName,
+            metadataNames: this.testReportsService.metadataNames,
+          } as View,
+        };
+        this.tabService.openNewTab(reportData);
+      },
+      error: () => catchError(this.errorHandler.handleError()),
+    });
+  }
+
+  getFullPaths(): void {
+    for (const report of this.reports) {
+      report.fullPath = '';
+      report.fullPath = report.path
+        ? `${report.path.replace(this.currentFilter, '')}${report.name}`
+        : `/${report.name}`;
```

Comment:
Why is `fullPath` assigned to an empty string first if it's always reassigned after?

Reply:
Added the change